pypesto/sample/util.py

```diff
@@ -0,0 +1,79 @@
+"""A set of helper functions"""
+import numpy as np
+import logging
+from typing import Tuple
+
+from ..result import Result
+from .diagnostics import geweke_test
+
+logger = logging.getLogger(__name__)
+
+
+def calculate_ci(
+        result: Result,
+        alpha: float = 0.95,
+        exclude_burn_in: bool = True,
+) -> Tuple[np.ndarray, np.ndarray]:
+    """
+    Calculate parameter confidence intervals based on MCMC samples.
+
+    Parameters
+    ----------
+    result:
+        The pyPESTO result object with filled sample result.
+    alpha:
+        Lower tail probability, defaults to 95% interval.
+
+    Returns
+    -------
+    lb, ub:
+        Bounds of the MCMC percentile-based confidence interval.
+    """
+    burn_in = 0
+    if exclude_burn_in:
+        # Check if burn in index is available
+        if result.sample_result.burn_in is None:
+            geweke_test(result)
+
+        # Get burn in index
+        burn_in = result.sample_result.burn_in
+
+    # Get converged parameter samples as numpy arrays
+    chain = np.asarray(result.sample_result.trace_x[0, burn_in:, :])
+
+    # Get percentile values corresponding to alpha
+    percentiles = 100 * np.array([(1-alpha)/2, 1-(1-alpha)/2])
+
+    # Get samples' upper and lower bounds
+    lb, ub = np.percentile(chain, percentiles, axis=0)
+
+    return lb, ub
+
+
+def calculate_prediction_profiles(
+        simulated_values: np.ndarray,
+        alpha: float = 0.95,
+) -> Tuple[np.ndarray, np.ndarray]:
+    """
+    Calculate prediction confidence intervals based on MCMC samples.
```

Comment:
```suggestion
    """Calculate prediction confidence intervals based on MCMC samples.
```

Reply:
I guess the discussion why this is called "_prediction_profiles", but above "_~~parameter~~_ci" has been had already?

Reply:
And also the discussion why this is called "confidence" interval here though sampling induces Bayesian setting, where thus "credible" intervals are obtained?

Reply:
I think this discussion was from before I was involved in this PR, but it's now reworded as suggested and refactored.
Better now?
Thoughts @elbaraim?

Reply:
Also, doesn't look like `calculate_prediction_profiles` (now renamed to `calculate_ci_mcmc_sample_prediction`) is used anywhere. And the only point where `calculate_ci` (now renamed to `calculate_ci_mcmc_sample`) is used is in a visualization method, which could be replaced with an ensemble prediction I guess (and then this `pypesto/sample/util.py` file can be deleted).

These methods were used in the `sampling_diagnostics.ipynb` before the ensemble predictions were implemented in pyPESTO.

Reply:
> And also the discussion why this is called "confidence" interval here though sampling induces Bayesian setting, where thus "credible" intervals are obtained?

fine for me to rename all to `credible` instead of `confidence`